File: 1078-remove-outermost-parentheses/remove-outermost-parentheses.cpp
class Solution {
public:
    string removeOuterParentheses(string s) 
    {
        string ans="";
        int depth=0;
        for(int i=0;i<s.length();i++)
        {
            if(s[i]=='(')
            depth++;
            else
            depth--;

            if(depth==0 || (depth==1 && s[i]=='('))
            continue;
            ans.push_back(s[i]);
        }
        return ans;
    }
};

File: 1078-remove-outermost-parentheses/remove-outermost-parentheses.java
class Solution {
    private boolean isDepthNonZero(int depth){
        return depth > 0;
    }
    public String removeOuterParentheses(String s) {
        int depth = 0;
        StringBuilder sb = new StringBuilder();
        for(char c : s.toCharArray()){
            if(c == '(') {
                if(isDepthNonZero(depth)){
                    sb.append(c);
                }
                depth++;
            }else {
                depth--;
                if(isDepthNonZero(depth)){
                    sb.append(c);
                }
            }
        }
        return sb.toString();
    }
}